wasp-ai/src/server/operations.ts

```diff
@@ -222,7 +209,46 @@ export const getStats = (async (_args, context) => {
 }) satisfies GetStats<{}>;
 
 export const getNumProjects = (async (_args, context) => {
-  const { Project } = context.entities;
-  const numProjects = await Project.count();
-  return numProjects;
+  return context.entities.Project.count();
 }) satisfies GetNumProjects<{}>;
+
+export const getProjectsByUser: GetProjectsByUser<void, Project[]> = async (_args, context) => {
+  if (!context.user) {
+    throw new HttpError(401, "Not authorized");
+  }
+  return await context.entities.Project.findMany({
+    where: {
+      user: {
+        id: context.user.id,
+      },
+    },
+  });
+};
+
+import { CheckIfUserStarredWasp } from "@wasp/queries/types";
+
+export const checkIfUserStarredWasp: CheckIfUserStarredWasp<{ username: string }, boolean> = async (
+  { username },
+  context
+) => {
+  if (!context.user) {
+    throw new HttpError(401, "Not authorized");
+  }
+  let status = false;
+  try {
+    const response = await fetch(`https://api.github.com/user/starred/wasp-lang/wasp`, {
+      method: "GET",
+      headers: {
+        Authorization: "Basic " + btoa(username + ":" + process.env.GITHUB_PERSONAL_ACCESS_TOKEN),
```

Comment:
So we will make this check every time user comes to the Mage? I wonder if we are going to hit some Github rate limits. Could we maybe cache it in some way? For example, if we realized they starred us, we store that into our database and don't ask them again? Yeah, maybe that would be a good way to do it? Sure, they could unstar us, but that is really rare and I don't we want to bother them in that case anyway, I would let them be.

Reply:
Ok this is irrevelant now.